examples/camera-app/linux/src/clusters/webrtc_provider/webrtc-provider-manager.cpp

```diff
@@ -228,10 +244,19 @@ CHIP_ERROR WebRTCProviderManager::SendAnswerCommand(uint16_t sessionId, const st
     };
 
     Messaging::ExchangeManager & exchangeMgr = Server::GetInstance().GetExchangeManager();
+    auto sessionHandle                       = exchangeMgr.GetSessionManager()->FindSecureSessionForNode(peerId);
```

Comment:
This is not the right way to get a session.  The right way is viea CASESessionManager, because the session might need to be established (async!) before it can be used.

I guess let's just do a followup for that, but please make sure it happens, and note that the async nature of that process will require a bunch of changes to this code.  But at least the cluster delegate API is OK now.

Reply:
https://github.com/project-chip/connectedhomeip/issues/38183

Does this behavior only apply when sending commands from the controller to the server? When the server sends a command to the controller, can a CASE session be automatically established? I don't believe the controller has MDNS broadcasting enabled. I'm also curious why existing implementations don't utilize these async methods.

Reply:
clarified offline

Reply:
> Does this behavior only apply when sending commands from the controller to the server?

No.

> I don't believe the controller has MDNS broadcasting enabled

If it wants things to send it commands not as part of existing interactions, it better have it enabled.

> I'm also curious why existing implementations don't utilize these async methods.

They totally do.